sagemaker-python-sdk/tensorflow_script_mode_training_and_serving/tensorflow_script_mode_training_and_serving.ipynb

```diff
@@ -0,0 +1,254 @@
+    "\n",
+    "The ``sagemaker.tensorflow.TensorFlow`` estimator handles locating the script mode container, uploading your script to a S3 location and creating a SageMaker training job. Let's call out couple important parameters here:\n",
+    "\n",
+    "* ``py_version`` is set to ``'py3'`` to indicate that we are using script mode since legacy mode supports only Python 2. Though Python 2 will be deprecated soon, you can use script mode with Python 2 by setting ``py_version`` to ``'py2'`` and ``script_mode`` to ``True``.\n",
+    "\n",
+    "* ``distributions`` is used to configure the distributed training setup. It's required only if you are doing distributed training either across a cluster of instances or across multiple GPUs. Here we are using parameter servers as the distributed training schema. SageMaker training jobs run on homogeneous clusters. To make parameter server more performant in the SageMaker setup we run a parameter server on every instance in the cluster, so there is no need to specify the number of parameter servers to launch. Script mode also supports distributed training with [Horovod](https://github.com/horovod/horovod). You can find the full documentation on how to configure ``distributions`` [here](https://github.com/aws/sagemaker-python-sdk/tree/master/src/sagemaker/tensorflow#distributed-training). \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "\n",
+    "mnist_estimator = TensorFlow(entry_point='mnist.py',\n",
+    "                             role=role,\n",
+    "                             train_instance_count=2,\n",
+    "                             train_instance_type='ml.p2.xlarge',\n",
+    "                             framework_version='1.12',\n",
+    "                             py_version = 'py3',\n",
+    "                             distributions = {'parameter_server': {'enabled': True}})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Calling ``fit``\n",
+    "\n",
+    "To start a training job, we call `estimator.fit(training_data_uri)`.\n",
+    "\n",
+    "You can pass an [``s3_input``](https://sagemaker.readthedocs.io/en/stable/session.html#sagemaker.session.s3_input) a class defined in SageMaker Python SDK which represents channel configurations for S3 data sources, a dictionary or an S3 location to the ``fit`` method. If a dictionary is used the keys, named channels, \n",
+    "have values pointing to the data location. `estimator.fit(training_data_uri)` downloads the container contains TensorFlow GPU binary built with Python3.6 to the SageMaker training job instances. If an S3 location is used the estimator will create a default channel named ``'training'`` which points to this S3 location.\n",
```

Comment:
this paragraph is still a bit unwieldy. I think focusing on which input type is used here and then just linking to the API docs for everything else would probably be easiest